pilot/pkg/model/validation.go

```diff
@@ -446,10 +446,6 @@ func validateServer(server *networking.Server) (errs error) {
 		errs = appendErrors(errs, fmt.Errorf("server config must contain at least one host"))
 	} else {
 		for _, host := range server.Hosts {
-			// short name hosts are not allowed in gateways
-			if host != "*" && !strings.Contains(host, ".") {
-				errs = appendErrors(errs, fmt.Errorf("short names (non FQDN) are not allowed in Gateway server hosts"))
-			}
```

Comment:
why remove this?

Reply:
This check fails for a host like  `ns1/*`, which is valid. The correct check for short hosts is done in `validateNamespaceSlashWildcardHostname`, just below.